packages/process/src/exec/posix.ts

```diff
@@ -59,22 +59,31 @@ export const createPosixProcess: CreateOSProcess = (command, options) => {
       };
 
       yield spawn(function*(task) {
+        yield label({ name: 'exec', state: 'running' });
         yield spawn(function*() {
-          let value: Error = yield once(childProcess, 'error');
+          yield label({ name: 'listen for error' });
+          let value: Error = yield withLabels(once(childProcess, 'error'), {
+            name: 'untilFirst(error)',
```

Comment:
I'm a bit confused by this naming scheme, it looks like `untilFirst` is a function which takes an error as an argument of `error`, but this doesn't really reflect the implementation. Would it make sense to name this something like `untilTermination` or `trapError` or just `trap` or something?

Reply:
Maybe we've already established this notation elsewhere? I'm a bit out of the loop at the moment.

Reply:
Yeah, we don’t really have good guidelines and conventions on how best to label yet. Perhaps we should come up with some